java/wavefrontsender.md

```diff
@@ -0,0 +1,212 @@
+# WavefrontSender
+
+## Table of Content
+
+* [Overview](#Overview)
+* [Option 1: Sending Data via the Wavefront Proxy](#Option-1-Sending-Data-via-the-Wavefront-Proxy)
+* [Option 2: Sending Data via Direct ingestion](#Option-2-Sending-Data-via-Direct-ingestion)
+* [Share a WavefrontSender](#Share-a-WavefrontSender)
+* [What's next](#What's-next)
+
+## Overview
+
+You can choose to send metrics, histograms, or trace data from your application to the Wavefront service using one of the following techniques:
+
+* Use a [**Wavefront proxy**](https://docs.wavefront.com/proxies.html), which then forwards the data to the Wavefront service. This is the recommended choice for a large-scale deployment that needs resilience to internet outages, control over data queuing and filtering, and more.
+* Use [**direct ingestion**](https://docs.wavefront.com/direct_ingestion.html) to send the data directly to the Wavefront service. This is the simplest way to get up and running quickly. 
+
+The `WavefrontSender` interface has two implementations. Instantiate the implementation that corresponds to your choice:
+
+* Option 1: [Create a `WavefrontProxyClient`](#Option-1-Sending-Data-via-the-Wavefront-Proxy) to send data to a Wavefront proxy.
+* Option 2: [Create a `WavefrontDirectIngestionClient`](#Option-2-Sending-Data-via-Direct-ingestion) to send data directly to a Wavefront service.
+
+## Option 1: Sending Data via the Wavefront Proxy
+
+>**Prerequisite**
+>Before your application can use a `WavefrontProxyClient`, you must [set up and start a Wavefront proxy](https://github.com/wavefrontHQ/java/tree/master/proxy#set-up-a-wavefront-proxy).
+
+When sending data via the Wavefront proxy, you need to create a `WavefrontProxyClient`. You need to build it with the information needed to send data to a Wavefront proxy, including:
+
+* The name of the host that will run the Wavefront proxy.
+* One or more proxy listening ports to send data to. The ports you specify depend on the kinds of data you want to send (metrics, histograms, and/or trace data). You must specify at least one listener port. 
+* Optional settings for tuning communication with the proxy.
```

Comment:
Optional configuration options? Is this talking about https://docs.wavefront.com/proxies_configuring.html or https://docs.wavefront.com/proxies_preprocessor_rules.html or both? Do we want a link?